clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -215,6 +414,155 @@ public static void writeUnsignedIntLE(byte[] buffer, int offset, int value) {
         buffer[offset]   = (byte) (value >>> 8 * 3);
     }
 
+    /**
+     * Write the given unsigned integer following the variable-length from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output. Since it the value is not negative Zig-zag is not used.
+     *
+     * @param value The value to write
+     * @param out The output to write to
+     */
+    public static void writeUnsignedVarInt(int value, DataOutput out) throws IOException {
+        while ((value & 0xffffff80) != 0L) {
+            out.writeByte((value & 0x7f) | 0x80);
+            value >>>= 7;
+        }
+        out.writeByte(value & 0x7f);
+    }
+
+    /**
+     * Write the given unsigned integer following the variable-length from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output. Since it the value is not negative Zig-zag is not used.
+     *
+     * @param value The value to write
+     * @param buffer The buffer to write to
+     */
+    public static void writeUnsignedVarInt(int value, ByteBuffer buffer) {
+        while ((value & 0xffffff80) != 0L) {
+            byte b = (byte) ((value & 0x7f) | 0x80);
+            buffer.put(b);
+            value >>>= 7;
+        }
+        buffer.put((byte) (value & 0x7f));
+    }
+
+    /**
+     * Write the given unsigned long following the variable-length from
+     * <a href="http://code.google.com/apis/protocolbuffers/docs/encoding.html">
+     * Google Protocol Buffers</a> into the output. Since it the value is not negative Zig-zag is not used.
+     *
+     * @param value The value to write
+     * @param out The output to write to
+     */
+    public static void writeUnsignedVarLong(long value, DataOutput out) throws IOException {
```

Comment:
I think we should introduce a `ByteUtils` class and move all the relevant methods there. Maybe we could do that in its own PR, which could be merged quickly independently of the message format changes?